src/main/java/com/onwelo/tinvalidator/validator/EEValidator.java

```diff
@@ -4,39 +4,22 @@
 
 public class EEValidator implements Validator {
 
-    private static final Pattern PATTERN = Pattern.compile("[1-6]\\d{2}[01]\\d[0-3]\\d[0-7]\\d{3}");
+    private static final Pattern PATTERN = Pattern.compile("\\d{9}");
 
     public Pattern getPattern(){
         return PATTERN;
     }
 
     public boolean computeControlSum(String tin) {
-        int[] weights = {1, 2, 3, 4, 5, 6, 7, 8, 9, 1};
+        int[] weights = {3, 7, 1, 3, 7, 1, 3, 7};
         try {
-            assert Integer.parseInt(tin.substring(3,5)) <= 12;
-            assert Integer.parseInt(tin.substring(3,5)) > 0;
-            assert Integer.parseInt(tin.substring(5,7)) <= 31;
-            assert Integer.parseInt(tin.substring(5,7)) > 0;
-            assert Integer.parseInt(tin.substring(7,10)) <= 710;
-            assert Integer.parseInt(tin.substring(7,10)) > 0;
+            assert Integer.parseInt(tin.substring(0, 2)) == 10;
 
             int sum = 0;
             for (int i = 0; i < weights.length; i++) {
                 sum += Integer.parseInt(tin.substring(i, i + 1)) * weights[i];
             }
-            if (sum % 11 != 10) {
-                return (sum % 11) == Integer.parseInt(tin.substring(10, 11));
-            }
-
-            int[] weights2 = {3, 4, 5, 6, 7, 8, 9, 1, 2, 3};
-            sum = 0;
-            for (int i = 0; i < weights2.length; i++) {
-                sum += Integer.parseInt(tin.substring(i, i + 1)) * weights2[i];
-            }
-            if (sum % 11 != 10) {
-                return (sum % 11) == Integer.parseInt(tin.substring(10, 11));
-            }
-            return 0 == Integer.parseInt(tin.substring(10, 11));
+            return (10 - (sum % 10)) % 10 == Integer.parseInt(tin.substring(8, 9));
```

Comment:
W algorytmie dla estoni nie widzę nigdzie modulo 10. Chyba że jest to jakiś sprytny inny zapis albo algorytm z innego dokumentu.

Reply:
tak, zastąpiłem `ceiling` modulo, efekt jest ten sam